src/GenAPI/Microsoft.DotNet.GenAPI/GenAPIApp.cs

```diff
@@ -0,0 +1,158 @@
+            {
+                Assemblies = assemblies;
+                AssemblyReferences = assemblyReferences;
+                OutputPath = outputPath;
+                HeaderFile = headerFile;
+                ExceptionMessage = exceptionMessage;
+                ExcludeAttributesFiles = excludeAttributesFiles;
+                IncludeVisibleOutsideOfAssembly = includeVisibleOutsideOfAssembly;
+            }
+
+            /// <summary>
+            /// The path to one or more assemblies or directories with assemblies.
+            /// </summary>
+            public string[] Assemblies { get; }
+
+            /// <summary>
+            /// Paths to assembly references or their underlying directories for a specific target framework in the package.
+            /// </summary>
+            public string[]? AssemblyReferences { get; }
+
+            /// <summary>
+            /// Output path. Default is the console. Can specify an existing directory as well and
+            /// then a file will be created for each assembly with the matching name of the assembly.
+            /// </summary>
+            public string? OutputPath { get; }
+
+            /// <summary>
+            /// Specify a file with an alternate header content to prepend to output.
+            /// </summary>
+            public string? HeaderFile { get; }
+
+            /// <summary>
+            /// Method bodies should throw PlatformNotSupportedException.
+            /// </summary>
+            public string? ExceptionMessage { get; }
+
+            /// <summary>
+            /// The path to one or more attribute exclusion files with types in DocId format.
+            /// </summary>
+            public string[]? ExcludeAttributesFiles { get; }
+
+            /// <summary>
+            /// Include all API's not just public APIs. Default is public only.
+            /// </summary>
+            public bool IncludeVisibleOutsideOfAssembly { get; }
+        }
+
+        /// <summary>
+        /// Initialize and run Roslyn-based GenAPI tool.
+        /// </summary>
+        public static void Run(Context context)
+        {
+            IAssemblySymbolLoader loader = new AssemblySymbolLoader(true);
+
+            if (context.AssemblyReferences is not null)
+            {
+                loader.AddReferenceSearchPaths(context.AssemblyReferences);
+            }
+
+            var intersectionFilter = new CompositeFilter()
+                .Add<ImplicitSymbolsFilter>()
+                .Add(new SymbolAccessibilityBasedFilter(context.IncludeVisibleOutsideOfAssembly));
+
+            if (context.ExcludeAttributesFiles != null)
+            {
+                intersectionFilter.Add(new AttributeSymbolFilter(context.ExcludeAttributesFiles));
+            }
+
+            var assemblySymbols = loader.LoadAssemblies(context.Assemblies);
+            foreach (var assemblySymbol in assemblySymbols)
+            {
+                if (assemblySymbol == null) continue;
```

Comment:
I would add a "// TODO: Add logging for the assembly symbol loading failure".

Reply:
added